src/vs/workbench/contrib/positronOutputWebview/browser/notebookOutputWebviewServiceImpl.ts

```diff
@@ -74,20 +74,31 @@ export class PositronNotebookOutputWebviewService implements IPositronNotebookOu
 	}
 
 	/**
-	 * Gather preferred renderer and the mime type renderer is for from an output message.
+	 * Get the preferred mime type and renderer for an output message.
 	 *
 	 * @param output An output messages to find renderers for.
 	 * @returns A renderer and the mime type it is preferred for along with the output message.
 	 */
-	private _findRendererForOutput(output: ILanguageRuntimeMessageWebOutput): MessageRenderInfo | undefined {
-		for (const mimeType in output.data) {
-			const renderer = this._notebookService.getPreferredRenderer(mimeType);
-			if (renderer) {
-				return { mimeType, renderer, output };
-			}
+	private _findRendererForOutput(output: ILanguageRuntimeMessageWebOutput, viewType?: string): MessageRenderInfo | undefined {
+		// Get the preferred mime type.
+		// This is the same logic used in CellOutputViewModel.resolveMimeTypes.
+		const mimeTypes = this._notebookService.getMimeTypeInfo(
+			viewType, undefined, Object.keys(output.data)
+		);
+		const pickedMimeType = mimeTypes.find(
+			mimeType => mimeType.rendererId !== RENDERER_NOT_AVAILABLE && mimeType.isTrusted
+		);
+		if (!pickedMimeType) {
+			return;
+		}
+
+		// Get the renderer for the preferred mime type.
+		const renderer = this._notebookService.getRendererInfo(pickedMimeType.rendererId);
+		if (!renderer) {
+			return;
```

Comment:
I can't believe this worked before. Thanks for fixing!

Reply:
I'm not sure the UI for doing it but I wonder in the future if we could do something like vscode notebooks where they let you choose a different renderer if desired.

Reply:
Agreed.